### ISSUE TITLE
Passing field names to @functor for typed struct gives constructor error

### ISSUE THREAD

**Issue by andrschl** [FluxML/Functors.jl#6]
I am not sure whether this is intended or not, but when trying to specify the trainable parameters in the following code a "no method matching error" for the constructor is returned. 
```
struct MyLayer{R, S, T}
    a::R
    b::S
    c::T
end
Flux.@functor MyLayer (a,b)
m = MyLayer(Dense(1,10,tanh), Dense(1,10,tanh), zeros(5))
Flux.destructure(m) # returns no method matching error for the constructor
```
In the last sentence of the corresponding section in the [documentation](https://fluxml.ai/Flux.jl/stable/models/advanced/) it is mentioned that a corresponding constructor has to exist. However, it is not clear to me how this constructor should look like.

**Comment by devmotion** [comment]
This issue seems to be the same as the one discussed in https://github.com/FluxML/Functors.jl/issues/3. You can check the comments therein for a solution to your problem: Instead of `Flux.@functor MyLayer (a,b)` you can define
```julia
function Functors.functor(::Type{<:MyLayer}, x)
    function reconstruct_MyLayer(xs)
        return MyLayer(xs.a, xs.b, x.c)
    end
    return (a = x.a, b = x.b), reconstruct_MyLayer
end
```

**Comment by CarloLucibello** [comment]
addressed in #7
